src/api/rootResolver.js

```diff
@@ -1,11 +1,29 @@
 import merge from 'lodash/merge'
-import eventsList from '../fixtures/events.json'
+import PublicDeployer from '@noblocknoparty/contracts/build/contracts/Deployer.json'
+import PrivateDeployer from '../build/contracts/Deployer.json'
+import eventsList from '../fixtures/events.json'  
+import {toHex, toWei} from 'web3-utils'
 import getEthers, { signer } from './ethers'
 import singleEventResolvers, {
   defaults as singleEventDefaults
 } from './resolvers/singleEventResolvers'
 import ensResolvers, { defaults as ensDefaults } from './resolvers/ensResolvers'
 
+const deployerAbi = PublicDeployer.abi
+const deployerContractAddresses = Object.assign(
+  {},
+  PrivateDeployer.networks,
+  PublicDeployer.networks
+)
+
+function getNetwork(){
+  return new Promise(function(resolve,reject){
+    window.web3.version.getNetwork(function(err, result){
```

Comment:
I am actually not sure if this works on mobile (status, cipher, trust wallet) which may inject web3 in different ways. Need to try out at the deployed env

Reply:
There is a way to do this from etherjs but [the way](https://twitter.com/makoto_inoue/status/1041323814980857856) etherjs is currently setup does not allow me to access to its `provider` object. Also we are using v3 but they already have v4. Should it switch to web3 1.0 rather than trying to upgrade to ethers v4? Having different ways to interact with web3 across different repos are really counter productive.

Reply:
I'm happy to migrate if we all want to use web3 1.0